api/resource/origin.go

```diff
@@ -55,3 +66,29 @@ func (origin *Origin) String() (string, error) {
 	anno, err := kyaml.Marshal(origin)
 	return string(anno), err
 }
+
+func OriginFromCustomPlugin(res *Resource) (*Origin, error) {
```

Comment:
Does this work correctly for custom generators defined inline in the Kustomization? Please add a test.

Reply:
It did not; thanks for the catch. I have fixed it and added tests.